src/panoramic/cli/command.py

```diff
@@ -251,3 +257,79 @@ def push(yes: bool = False, target_dataset: Optional[str] = None, diff: bool = F
             except Exception as e:
                 bar.write(f'Error: Failed to execute action {action.description}:\n  {str(e)}')
         bar.write(f'Updated {bar.total} models')
+
+
+def detect_joins(target_dataset: Optional[str] = None, diff: bool = False, overwrite: bool = False, yes: bool = False):
+    company_slug = get_company_slug()
+    echo_info('Loading local state...')
+    local_state = get_local_state(target_dataset=target_dataset)
+
+    if local_state.is_empty:
+        echo_info('No datasets to detect joins on')
+
+    join_detector = JoinDetector(company_slug=company_slug)
+    join_detector.fetch_token()
+
+    models_by_virtual_data_source: Dict[str, Dict[str, PanoModel]] = defaultdict(dict)
+    for model in local_state.models:
+        # Prepare a mapping for a quick access when reconciling necessary changes later
+        models_by_virtual_data_source[model.virtual_data_source][model.model_name] = model
+
+    actions_list = ActionList(actions=[])
+
+    with tqdm(list(local_state.data_sources)) as bar:
+        for dataset in bar:
+            try:
+                echo_info(f'Detecting joins for dataset {dataset.dataset_slug}')
```

Comment:
```suggestion

```

Reply:
Not using `bar.write` but also not strictly necessary

Reply:
Fixed in 742d698.